tests/TestDialogs/PassiveTestDialog.php

```diff
@@ -29,6 +29,7 @@ public function step2(Update $update): void
 
     public function step3(Update $update): void
     {
+        $this->jump("step2");
```

Comment:
Honestly, I'm not really happy about making this test class specific for this case. It will make this test Tet class full of magic and not really reusable for further changes. is it safe to remove this line?

Reply:
This line is for test of afterLastStep call, when dialog jumps from the last step. In this case afterLastStep should be also called, i think (BTW: I do not imagine the case, where beforeFirstStep and/or afterLastStep functions may be used, so this call is still a question). I agree with your conclusion. But, we can remove this line only with make of another test. What may it be?
1) Ideally to do like in DialogTest.php, where you initiate dialog instance with steps at a place. I tried this approach, but failed. Error: can not serialize object by anonymous user (smth like that) when "activate" function called. I didn't find solution for that.
2) Also it is the way to make FakeDialog file like FakeBot for similar small tests (and add cases there for different tests), but i didn't understand how to add function there.
3) Making another test dialog just for this test seems bad.

BTW: I build telegram bots from 2018. It is like my main job. But i do it with some no code service (Pipe.bot - like Manychat or Chatfuel) with adding of php scripts for advanced tasks. Now i'd like to work without any no code service. So i'm here and very interested in some packages like this and sdk) I want to say, i have a lot of experience about dialog cases and tg api, but with notice, that it was partially with nocode service. As about my php knowledge. I integrated almost all Ukrainian payment systems to my chatbots, using plain php and webhooks. I'm like juniour, but with strong university dev background of 2001-2007 ))) Probably it will be more quick and comfortable to communicate in tg? It will be for me for sure. My contact: @DimaOdnokoz there.

Reply:
ok, let's merge is as it, I'll reply later. Green CI is more important than clean tests